sql/transaction-isolation.md

```diff
@@ -0,0 +1,75 @@
+---
+title: TiDB Transaction Isolation Levels
+category: user guide
+---
+
+# TiDB Transaction Isolation Levels
+
+Transaction isolation is one of the foundations of database transaction processing. Isolation is the I in the acronym ACID (Atomicity, Consistency, Isolation, Durability), which represents the isolation property of database transactions.
+
+The SQL-92 standard defines four levels of transaction isolation: Read Uncommitted, Read Committed, Repeatable Read and Serializable. See the following table for details:
+
+| Isolation Level  | Dirty Read   | Nonrepeatable Read | Phantom Read          | Serialization Anomaly |
+| ---------------- | ------------ | ------------------ | --------------------- | --------------------- |
+| Read Uncommitted | Possible     | Possible           | Possible              | Possible              |
+| Read Committed   | Not possible | Possible           | Possible              | Possible              |
+| Repeatable Read  | Not possible | Not possible       | Not possible in  TiDB | Possible              |
+| Serializable     | Not possible | Not possible       | Not possible          | Not possible          |
+
+TiDB offers two transaction isolation levels: Read Committed and Repeatable Read.
+
+TiDB uses the [Percolator transaction model](https://research.google.com/pubs/pub36726.html). A global read timestamp is obtained when the transaction is started, and a global commit timestamp is obtained when the transaction is committed. The execution order of transactions is confirmed based on the timestamps. To know more about the implementation of TiDB transaction model, see [MVCC in TiKV](https://pingcap.com/blog/2016-11-17-mvcc-in-tikv/).
+
+Use the following command to set the level of transaction isolation:
+
+```
+SET SESSION TRANSACTION ISOLATION LEVEL [read committed|repeatable read]
+```
+
+## Repeatable Read
+
+Repeatable Read is the default transaction isolation level in TiDB. The Repeatable Read isolation level only sees data committed before the transaction began, and it never sees either uncommitted data or changes committed during transaction execution by concurrent transactions. However, the transaction statement does see the effects of previous updates executed within its own transaction, even though they are not yet committed.
```

Comment:
Repeatable Read is the default transaction isolation level in TiDB. The Repeatable Read isolation level only sees data committed before the transaction "begins"